core/vm/contracts.libevm.go

```diff
@@ -89,20 +89,26 @@ func (t CallType) OpCode() OpCode {
 }
 
 // run runs the [PrecompiledContract], differentiating between stateful and
-// regular types.
-func (args *evmCallArgs) run(p PrecompiledContract, input []byte, suppliedGas uint64) (ret []byte, remainingGas uint64, err error) {
-	if p, ok := p.(statefulPrecompile); ok {
-		return p(args.env(), input, suppliedGas)
+// regular types, updating `gasRemaining` in the stateful case.
```

Comment:
Nit just for expliciteness' sake / it would had helped my understanding 😉 

```suggestion
// regular types, updating the receiving args `gasRemaining` field in
// the stateful precompile case.
```

Reply:
I've gone with this instead:

```go
// run runs the [PrecompiledContract], differentiating between stateful and
// regular types, updating `args.gasRemaining` in the stateful case.
```

I find it's clearer / easier to use code instead of English to communicate things like this.